### PR TITLE
:seedling: bump k8s, helm-operator-plugins, of/operator-registry, and of/api

### PR DESCRIPTION
**Pull request by joelanford**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,121 +11,126 @@ require (
...
-	k8s.io/kube-aggregator v0.26.10
-	k8s.io/utils v0.0.0-20230220204549-a5ecb0141aa5
+	helm.sh/helm/v3 v3.13.3
+	k8s.io/api v0.28.5
+	k8s.io/apiextensions-apiserver v0.28.5
+	k8s.io/apimachinery v0.28.5
+	k8s.io/cli-runtime v0.28.5
+	k8s.io/client-go v0.28.5
+	k8s.io/component-base v0.28.5
+	k8s.io/kube-aggregator v0.28.5
```
Comment (everettraven) [inline go.mod:29]: We should probably bump the KinD version to v0.20.0 and use a 1.28 based node image to ensure we are testing against a 1.28 cluster in e2e

Comment (joelanford) [reply go.mod:29]: sgtm

Comment (joelanford) [reply go.mod:29]: 🤔 Something about this breaks the e2e test where we pull from a private registry. Investigating.

**Comment by joelanford** [comment]
Fixes #799

File: go.mod (hunk 1 above)
Comment (joelanford) [reply go.mod:29]: I'll try this again after I rebase, but I'm tempted to separate this out into a separate PR if I can't get it working immediately.

Comment (joelanford) [reply go.mod:28]: https://github.com/operator-framework/rukpak/issues/803

File: cmd/crdvalidator/handlers/crd.go [hunk 2]
```diff
@@ -74,12 +75,6 @@ func (cv *CrdValidator) Handle(ctx context.Context, req admission.Request) admis
 	return admission.Allowed("")
 }
 
-// InjectDecoder injects a decoder for the CrdValidator.
-func (cv *CrdValidator) InjectDecoder(d *admission.Decoder) error {
-	cv.decoder = d
-	return nil
-}
-
```
Comment (everettraven) [inline cmd/crdvalidator/handlers/crd.go:82]: Any particular reason for this removal? I noticed that this made the go-apidiff check unhappy

File: .bingo/variables.env [hunk 3]
```diff
@@ -12,7 +12,7 @@ BINGO="${GOBIN}/bingo-v0.8.0"
 
 CONTROLLER_GEN="${GOBIN}/controller-gen-v0.12.0"
```
Comment (varshaprasad96) [inline .bingo/variables.env:13]: Do we also want to bump this to 0.13.0: https://github.com/kubernetes-sigs/controller-tools/blob/v0.13.0/go.mod

**Review by everettraven — APPROVED** [review]
Aside from the go-apidiff failure this LGTM. If that is expected then 👍

**Review by varshaprasad96 — COMMENTED** [review]
Just one question regarding controller-gen bump.
Rest looks good to me, /lgtm from my end

File: .bingo/variables.env (hunk 3 above)
Comment (joelanford) [reply .bingo/variables.env:13]: I'll do this in a follow-up. Thanks for pointing it out!

File: cmd/crdvalidator/handlers/crd.go (hunk 2 above)
Comment (joelanford) [reply cmd/crdvalidator/handlers/crd.go:82]: That was the old way controller-runtime could inject a Decoder. That interface was removed, and the function is no longer useful.

I updated the code to set the decoder in the constructor function instead.

**Pull request merged**